### PR TITLE
Cleanup unused metadata in agent

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Review by nmittler — APPROVED** [review]

**Comment by howardjohn** [comment]
/retest

to see if its a flake

**Comment by howardjohn** [comment]
Oops I thought https://github.com/istio/istio/pull/22200 was merged already... this depends on that. I pull in those changes @rshriram

File: manifests/gateways/istio-ingress/templates/deployment.yaml [hunk 1]
```diff
@@ -180,19 +178,6 @@ spec:
...
-          {{- if $.Values.global.mtls.auto }}
-          - name: ISTIO_AUTO_MTLS_ENABLED
-            value: "true"
-          {{- end }}
-          - name: ISTIO_META_POD_NAME
-            valueFrom:
-              fieldRef:
-                apiVersion: v1
-                fieldPath: metadata.name
-          - name: ISTIO_META_CONFIG_NAMESPACE
```
Comment (rshriram) [inline manifests/gateways/istio-ingress/templates/deployment.yaml]: So we have fully switched to NAMESPACE ?

Comment (rshriram) [reply manifests/gateways/istio-ingress/templates/deployment.yaml]: cc @vikaschoudhary16 (fyi)

File: pilot/pkg/networking/core/v1alpha3/gateway.go [hunk 2]
```diff
@@ -419,7 +419,7 @@ func buildGatewayListenerTLSContext(
 		},
 	}
 
-	if enableIngressSds && server.Tls.CredentialName != "" {
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/gateway.go]: didn't I have a PR for this?

**Review by rshriram — APPROVED** [review]

**Pull request merged**